pkg/contexts/credentials/builtin/npm/identity/identity.go

```diff
@@ -65,3 +68,47 @@ func GetCredentials(ctx cpi.ContextProvider, repoUrl string, pkgName string) com
 	}
 	return credentials.Properties()
 }
+
+// BearerToken retrieves the bearer token for the given repository URL and package name.
+// Either it's setup in the credentials or it will login to the registry and retrieve it.
+func BearerToken(ctx cpi.ContextProvider, repoUrl string, pkgName string) (string, error) {
```

Comment:
I think to stay uniform with the existing packages, we should also move `BearerToken()` and `Authorize()` to the npm login file.

Reply:
but that would lead to cyclic dependencies between the `npm.identity` and the generic npm package

Reply:
done